ext/gvltools/instrumentation.c

```diff
@@ -163,11 +199,15 @@ void Init_instrumentation(void) {
 
     VALUE rb_mGlobalTimer = rb_const_get(rb_mGVLTools, rb_intern("GlobalTimer"));
     rb_define_singleton_method(rb_mGlobalTimer, "reset", global_timer_reset, 0);
-    rb_define_singleton_method(rb_mGlobalTimer, "monotonic_time", global_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mGlobalTimer, "monotonic_time", global_wait_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mGlobalTimer, "monotonic_wait_time", global_wait_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mGlobalTimer, "monotonic_hold_time", global_hold_timer_monotonic_time, 0);
 
     VALUE rb_mLocalTimer = rb_const_get(rb_mGVLTools, rb_intern("LocalTimer"));
     rb_define_singleton_method(rb_mLocalTimer, "reset", local_timer_reset, 0);
-    rb_define_singleton_method(rb_mLocalTimer, "monotonic_time", local_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mLocalTimer, "monotonic_time", local_wait_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mLocalTimer, "monotonic_wait_time", local_wait_timer_monotonic_time, 0);
+    rb_define_singleton_method(rb_mLocalTimer, "monotonic_hold_time", local_hold_timer_monotonic_time, 0);
```

Comment:
So I know this isn't working as you intend yet, and I'm not yet sure it's a feature I want to include.

But the current design is that each counter is its own class and expose it's counter via `monotonic_time` or `count`.